backends/s3/marker.go

```diff
@@ -14,7 +14,7 @@ import (
 // anything and returns no error.
 func (b *Backend) setMarker(ctx context.Context, name, etag string, isDel bool) error {
 	if !b.opt.UseUpdateMarker {
-		return nil
+		return ctx.Err()
```

Comment:
I am struggling a little to understand why ctx.Err() is returned here. The doc above says:
// In case the UseUpdateMarker option is false, this function doesn't do
// anything and returns no error

so returning nil seems to better match that description?